### DISCUSSION TITLE
How to build

### DISCUSSION THREAD

**Discussion by buhtz** [ximion/appstream#585] [Q&A] [answered]
I got errors like this

https://www.freedesktop.org/software/appstream/docs/chap-Validation.html#asv-cid-maybe-not-rdns

But I can not find a good explanation how to build the rdns. The rdns from the generator is not good enough because it include names of code hosters (e.g. "github" for Microsoft GitHub) which makes absolute no sense to me because code hosters are not a persistent element.

**Comment by ximion** [comment]
Well, the rDNS is created from the website of your application. If that is on GitHub/GitLab/etc, then that's the ID you need. 
You can find more details at https://www.freedesktop.org/software/appstream/docs/chap-Metadata.html#tag-id-generic, but the MetaInfo Creator will produce a sensible ID.
If you have an own homepage for your app, it is likely a better idea to use that instead.

**Reply by buhtz** [reply]
But a website (even my own or a code hoster) is not persistent.
Should be an rDNS be persistent and unique?

Is there a rule against `org.bit-team.backintime.gui` ?

**Reply by ximion** [reply]
Nothing is persistent. Rocks crumble, mountains are razed, water evaporates, the atmosphere dissipates into space and eventually the universe will end.
But websites are reasonably persistent for this usecase and the DNS guarantees that the namespaces are kept unique(ish).
If you really move your VCS or domain at a point, you can use the `replaces` tag to move, but you should avoid renaming the ID as much as possible, as it identifies your app and auxiliary data may be tied to it (like its reviews, for example).

> Is there a rule against org.bit-team.backintime.gui ?

As long as you or your org owns the domain http://bit-team.org/ this is fine. You should consider replacing the dash with an underscore though (for harmonization reasons with DBUs).

**Reply by buhtz** [reply]
bit-team.org is not an existing domain.
The question is if it is mandatory that this has to be a domain or if it is optional.
I want to avoid using "github" as part of my ID.

**Reply by ximion** [reply, marked as answer]
It is required that it is an existing domain - otherwise the DNS can't ensure uniqueness.
You can use another domain, but it still has to reliably identify your project somehow (you can use GitLab for example, or any other open-source site... Or a blog at Wordpress, etc, whatever you like as long as the TLD exists and the parts after the TLD identify your project reliably).

For example, if AppStream wasn't a Freedesktop.org project (with an `org.freedesktop.appstream` ID) it could be `io.github.ximion.appstream`.
